File: IPRelated/IPManager.swift
//
//  IPManager.swift
//  SwiftiDate
//
//  Created by 游哲維 on 2025/3/11.
//

import Foundation

class IPManager {
    // 採用單例模式，方便在 App 中隨時調用
    static let shared = IPManager()
    
    private init() {}
    
    // 這個方法會呼叫外部 API 來取得對外 IP
    func fetchPublicIP(completion: @escaping (String?) -> Void) {
        guard let url = URL(string: "https://api.ipify.org?format=json") else {
            completion(nil)
            return
        }
        let task = URLSession.shared.dataTask(with: url) { data, response, error in
            guard
                let data = data,
                error == nil,
                let jsonObject = try? JSONSerialization.jsonObject(with: data, options: []),
                let json = jsonObject as? [String: Any],
                let ip = json["ip"] as? String
            else {
                completion(nil)
                return
            }
            completion(ip)
        }
        task.resume()
    }
}

File: flutter/swiftidate/ip_manager.dart
import 'dart:convert';
import 'package:http/http.dart' as http;

class IPManager {
  // 單例模式
  static final IPManager shared = IPManager._internal();
  IPManager._internal();

  /// 取得對外 IP，若成功回傳 IP 字串，否則回傳 null
  Future<String?> fetchPublicIP() async {
    try {
      final response = await http.get(Uri.parse("https://api.ipify.org?format=json"));
      if (response.statusCode == 200) {
        final jsonResponse = jsonDecode(response.body);
        return jsonResponse["ip"] as String?;
      } else {
        print("Error: HTTP ${response.statusCode}");
        return null;
      }
    } catch (e) {
      print("Error fetching public IP: $e");
      return null;
    }
  }
}
